Add pryrc overrides (#2)
firstdraft/appdev_support: README.md

  ```irb
  pry(main)> Todo.all
  => Todo::ActiveRecord_Relation (array with 1 Todo instance inside)
  ```
With the value `:full`
- `ActiveRecord::Relation` objects are displayed like this:
  ```irb
  pry(main)> Todo.all
  Todo Count (0.3ms)  SELECT COUNT(*) FROM "todos"
  Todo Load (0.2ms)  SELECT "todos".* FROM "todos"
  +----+-------+------+---------------------------------------------+
  | id | title | body | {:header_frequency=>10, :border=>:markdown} |
  +----+-------+------+---------------------------------------------+
  |  1 | test  | nope |                                             |
  +----+-------+------+---------------------------------------------+

    Todo Count (0.3ms)  SELECT COUNT(*) FROM "todos"
  => Todo::ActiveRecord_Relation (array with 1 Todo instance inside)
  ```
- `Class` objects are displayed like this:
  ```irb
  pry(main)> Todo
  => Todo(id: integer, title: string, body: text, created_at: datetime, updated_at: datetime)

  The Todo class itself.
  ```
- All other objects are displayed like this:
  ```irb
  pry(main)> "hello, world"
  => "hello, world"

  An instance of the String class.
  ```

> ## Configuration

Reviewer: I'm not sure if there's some best practice being followed, but should this section go below Installation? It took me a moment to figure out how to actually setup this Gem with it down here.

Author: I don't think I intentionally was following some convention when adding this section, but I do see some popular gems add configuration after usage.

> Add an initializer:

Reviewer: It took me a few to understand how this is setup. Partly just due to me being new to the dirty details of Rails configuration. But maybe you want to be explicit here and say (this is how I set it up, maybe there is a more correct way)
```suggestion
In your Rails application, add an initializer `config/initializers/appdev_support.rb`, and put the following code in it:
```

Reviewer 2: Perhaps a middle path: something like,

"Add an initializer to your application that contains the following:

...

In Rails, that would mean creating a file in the `config/` folder called e.g. `appdev_support.rb`."

> # AppdevSupport

Reviewer 2: Actually I just realized the class itself is called `AppdevSupport`, so nevermind me.

> For Rails gems **only**.

Author: Agreed. I think I'll save the unrelated README changes to a separate PR  or commit.

TODO: Delete this and the text above, and describe your gem

## Installation

Add this line to your application's Gemfile:

```ruby
gem 'appdev_support', git: "https://github.com/firstdraft/appdev_support"
```

And then execute:

    $ bundle install

Or install it yourself as:

    $ gem install appdev_support

Run

```rb
AppdevSupport.init
```
to load the default settings.

Create an [iniliazer](#configuration) file for customization.

## Usage

Once installed, you can do things like:

```ruby
# in app/controllers/events_controller.rb
class EventsController < ApplicationController 
  def index
    cookies.store(:theme, "dark")
    cookies.fetch(:theme)
    session.fetch(:user_id)
  end
end
```

and `session.fetch` will raise a `KeyNotFound` Exception.

`.fetch` can be called on `session`, `cookies`, and any `ActiveRecord` object. For an `ActiveRecord` object, you can `.fetch` a column.

```ruby
Event.fetch(:starts_on) # => "Mon, 23 Mar 2020 23:25:48"
```

Similarly, `.store` can be called on `session` and `cookies` with the expected behavior. 

```ruby
session.store(:user_id, 1)
```

Displaying an `ActiveRecord:Relation` in a View file will display:

```html

> ActiveRecord:Relation:Event (array with 4 Event instances inside)

Reviewer 2: ```suggestion
Event::ActiveRecord_Relation (array with 4 Event instances inside)

```

```

> You can call `.at` on an `ActiveRecord:Relation` instead of just `[]` to mirror how Arrays work.

Reviewer 2: ```suggestion
You can call `.at` on an `ActiveRecord:Relation` instead of just `[]` to mirror how `Array`s work.

```

With the value `:base`:
- `ActiveRecord::Relation` objects are displayed like this:
  ```irb
  pry(main)> Todo.all
  => Todo::ActiveRecord_Relation (array with 1 Todo instance inside)
  ```
With the value `:full`
- `ActiveRecord::Relation` objects are displayed like this:
  ```irb
  pry(main)> Todo.all
  Todo Count (0.3ms)  SELECT COUNT(*) FROM "todos"
  Todo Load (0.2ms)  SELECT "todos".* FROM "todos"
  +----+-------+------+---------------------------------------------+
  | id | title | body | {:header_frequency=>10, :border=>:markdown} |
  +----+-------+------+---------------------------------------------+
  |  1 | test  | nope |                                             |
  +----+-------+------+---------------------------------------------+

    Todo Count (0.3ms)  SELECT COUNT(*) FROM "todos"
  => Todo::ActiveRecord_Relation (array with 1 Todo instance inside)
  ```
- `Class` objects are displayed like this:
  ```irb
  pry(main)> Todo
  => Todo(id: integer, title: string, body: text, created_at: datetime, updated_at: datetime)

  The Todo class itself.
  ```
- All other objects are displayed like this:
  ```irb
  pry(main)> "hello, world"
  => "hello, world"

>   An instance of the String class.

Reviewer 2: I wonder if output like this would be clearer:

```
=> An instance of the String class:

"hello, world"
  
```

Author: @raghubetina to be consistent with this change, should ActiveRecord::Relation and ActiveRecord objects print their type first as well?

Reviewer 2: @jelaniwoods Yeah, I kinda like doing that for everything. The hash rocket before the return value has always confused a significant number of students. Then the actual value itself can be on its own line. Do you concur?

Reviewer 2: @jelaniwoods @bpurinton

Q: What do you think is better for learning between:

```
=> The String class itself:

String

```

and:

```
=> An instance of the Class class:

String

```

Author: > The hash rocket before the return value has always confused a significant number of students at the very beginning.

@raghubetina 

It's hard for me to imagine how I would feel about this if I _didn't_ know anything about programming 🤔. It feels a little uncomfortable because:
1. I'm so used to expecting the output to be after the Hash rocket and not below it
2. This behavior won't be in third party Ruby resources that we recommend when students need more practice (Treehouse, Exercism)

If hash rocket return values are confusing, do you think it would be worth it to have more content/practice around it? If the Ruby intro content is already being updated a lot, so I don't imagine it'd be too difficult to talk more about it.

> What do you think is better for learning between:

I'm leaning towards the first approach. I think it should immediately clear what type the object is when printing. While "instance of the class Class" isn't _wrong_, it's not as clear as saying "The class itself".
